xcm/xcm-builder/src/location_conversion.rs

```diff
@@ -147,6 +147,35 @@ impl<Network: Get<Option<NetworkId>>, AccountId: From<[u8; 20]> + Into<[u8; 20]>
 	}
 }
 
+/// Tries to convert global consensus parachain to accountId.
+///
+/// (E.g.: can be used for sovereign account conversion)
+pub struct GlobalConsensusParachainConvert<AccountId>(PhantomData<AccountId>);
+impl<AccountId: From<[u8; 32]> + Clone> Convert<MultiLocation, AccountId>
+	for GlobalConsensusParachainConvert<AccountId>
+{
+	fn convert_ref(location: impl Borrow<MultiLocation>) -> Result<AccountId, ()> {
+		match location.borrow() {
+			MultiLocation {
+				interior: X2(GlobalConsensus(network), Parachain(para_id)), ..
```

Comment:
This line is rather dirty. Though probably fine since it's unlikely any non-top-level parent nodes of the context location would contain this pattern, it's still at best careless and at worst a little bit dangerous.

`location` should first be converted into a Universal Location by interpreting it from the perspective of the local context. For this, you'll want to pass in the chain's `UniversalLocation` and `reanchor` it with `location` to give you `location` seen as a Universal Location, which basically means an `InteriorMultiLocation` value, which has no `parents` by definition.

`parents` should be specified according to the local chain's location in oniversal consensus. Alternatively (and probably more easily) pass in

Reply:
thank you,
I added `UniversalLocation` and I tried to reuse `universal_exports::ensure_is_remote` here,
there is no `parents` handling and everything looks much more clear now, 

converter now works only for `parents: 2`,
e.g.: `MultiLocation::new(2, X2(GlobalConsensus(networkId)), Parachain(paraId)))`,
which could be ok (it is exactly my use-cases).

or should I better not use `universal_exports::ensure_is_remote` and use `reanchor` with some `parents` number checking/handling?

Reply:
It looks to me that a simpler solution would be to specify the `parents` field as `2` here, rather than calling the `ensure_is_remote` function. Am I missing something here?

Reply:
@gavofyork @KiChjang 

my scenario is:

1. in Polkadot's Statemint runtime with UniversalLocation:
```
pub UniversalLocation: InteriorMultiLocation = X2(GlobalConsensus(Polkadot), Parachain(1000));
```

2. in Polkadot's Statemint runtime's XcmExecutor we receive bridged xcm call from Kusama/Statemine which starts with:
```
				UniversalOrigin(GlobalConsensus(Kusama)),
				DescendOrigin(X1(Parachain(1001))),
                                ...
```

3. so, the origin/location, which comes to the converter is like:
```
let location = MultiLocation { parents: 2, interior: X2(GlobalConsensus(Kusama), Parachain(1001)) };
```

4. so, [`ensure_is_remote` here](https://github.com/paritytech/polkadot/pull/7016/files#diff-b210cb75f8786021ba64cb7a2377778668ce90a2c306f6a1a1b822a2bba61cfaR285-R289) does exactly what I need/expect:
```
let devolved = ensure_is_remote(
        X2(GlobalConsensus(Polkadot), Parachain(1000)),
        MultiLocation { parents: 2, interior: X2(GlobalConsensus(Kusama), Parachain(1001)) }
);
let (remote_network, remote_location) = devolved;
// remote_network = Kusama
// remote_location = Parachain(1001)
```

5. so then the accountId is calculated as `(prefix, Kusama, 1001).using_encoded(blake2_256)` - works ok

---
so, I am not sure, how exactly should I handle `parents` here or if I need to handle `parents` somehow? 
`UniversalLocation` is `InteriorMultiLocation`, so no parents.
The input location is `MultiLocation`, so I could use `reanchor`, but I dont know what `???target???` should I use e.g:
```
location.reanchor(???target???, UniversalLocation::get())?;
```

Reply:
> It looks to me that a simpler solution would be to specify the `parents` field as `2` here, rather than calling the `ensure_is_remote` function. Am I missing something here?

No - that would make an assumption about where in consensus the local chain is (i.e. that it is a parachain).